src/checker/Pulse.Checker.Prover.Match.fst

```diff
@@ -247,42 +244,56 @@ let try_solve_uvars (g:env) (uvs:env { disjoint uvs g }) (p q:term)
     let q_names = freevars q in
     L.fold_left (fun (ss:(ss:PS.ss_t { PS.dom ss `Set.subset` freevars q })) (x, t) ->
       let nv_view = R.inspect_namedv x in
-      let topt = readback_ty t in
-      match topt with
-      | Some t ->
-        if Set.mem nv_view.uniq q_names &&
-           not (Set.mem nv_view.uniq (PS.dom ss))
-        then begin
-          let ss_new = PS.push ss nv_view.uniq t in
-          assert (nv_view.uniq `Set.mem` freevars q);
-          assert (PS.dom ss `Set.subset` freevars q);
-          assume (PS.dom ss_new `Set.subset` freevars q);
-          ss_new
-        end
-        else ss
-      | None -> ss
+      if Set.mem nv_view.uniq q_names &&
+         not (Set.mem nv_view.uniq (PS.dom ss))
+      then begin
+        let ss_new = PS.push ss nv_view.uniq t in
+        assert (nv_view.uniq `Set.mem` freevars q);
+        assert (PS.dom ss `Set.subset` freevars q);
+        assume (PS.dom ss_new `Set.subset` freevars q);
+        ss_new
+      end
+      else ss
     ) ss l
 
+let eq_tm_unascribe (g:env) (p q:term)
```

Comment:
vprop matching is now working on F* reflection terms. In some cases, we do try to match `p <: vprop` with `p` in the context. If we just did syntactic matching, the ascriptions come in the way.

To get around this, the function eq_tm_unascribe unascribes and then does eq_tm, it returns an equiv proof for the two terms, by using magic () for the equiv token.

Going forward, we could move it to calling F* Rel (in some restricted config, no delta, no smt etc.), and get the token from there.